File: ga/4omini_result_2024_7/result/macro_func_mini/20_macro_func.cpp
use std::io;

macro_rules! cppmax {
    ($x:expr, $y:expr) => {
        if ($x + $y) < 5 {
            $y
        } else {
            $x
        }
    };
}

fn main() {
    let mut input = String::new();

    println!("Enter two integers:");
    io::stdin().read_line(&mut input).expect("Failed to read line");

    let numbers: Vec<&str> = input.trim().split_whitespace().collect();
    let a: i32 = numbers[0].parse().expect("Invalid input for a");
    let b: i32 = numbers[1].parse().expect("Invalid input for b");

    let result = cppmax!(a, b);
    println!("Result: {}", result);
}

File: ga/gpt3_5_result_2024_6/result/macro_func_gpt/20_macro_func.rs
use std::io;

fn cppmax(x: i32, y: i32) -> i32 {
    if x + y < 5 {
        return y;
    } else {
        return x;
    }
}

fn main() {
    let mut input = String::new();

    println!("Enter two integers:");
    io::stdin().read_line(&mut input).expect("Failed to read input");
    let numbers: Vec<i32> = input.trim().split_whitespace()
        .filter_map(|x| x.parse().ok())
        .collect();

    let a = numbers[0];
    let b = numbers[1];

    let result = cppmax(a, b);
    
    println!("Result: {}", result);
}